pipeline/pipelines/batch_generation_pipeline.py

```diff
@@ -0,0 +1,133 @@
+        for single_flow in self.get_language_api_flows(**kwargs):
+            batch_flow.add(single_flow)
+
+        # TODO(michaelbausor): add task to create Github PR once staging repo
+        # is public on Github
+        final_tasks = []
+
+        flow = linear_flow.Flow('BatchGenerationFlow')
+        flow.add(batch_flow)
+        flow.add(*task_utils.instantiate_tasks(final_tasks, kwargs))
+
+        return flow
+
+    def get_language_api_flows(self, apis, languages, api_config_pattern,
+                               artman_language_yaml, **kwargs):
+        # Remove language arg from kwargs - we don't want it to override
+        # the language set in _get_configs
+        del kwargs['language']
+
+        repo_root = kwargs['repo_root']
+        api_config_yamls = _get_api_configs(api_config_pattern, apis)
+
+        for api_kwargs in _get_configs(
+                api_config_yamls, languages, artman_language_yaml, repo_root):
+
+            api_kwargs.update(kwargs)
+
+            tasks = self.get_pipeline_tasks_func(**api_kwargs)
+            tasks += task_utils.instantiate_tasks(
+                [batch_tasks.BatchOutputDirTask,
+                 batch_tasks.BatchCleanTask,
+                 batch_tasks.BatchCopyTask],
+                api_kwargs)
+
+            single_flow = linear_flow.Flow('SingleLanguageApiFlow')
+            single_flow.add(*tasks)
+            yield single_flow
+
+
+def _get_configs(api_config_yamls, languages, artman_language_yaml, repo_root):
+    for language in _get_languages(artman_language_yaml, languages):
+        lang_config = _load_artman_config(artman_language_yaml,
+                                          language,
+                                          repo_root)
+        lang_config['language'] = language
+        # TODO(michaelbausor): change default to True once most API and
+        # language combinations are supported
+        if not lang_config.get('enable_batch_generation', False):
+            continue
+        for api_config_yaml in api_config_yamls:
+            api_config = _load_artman_config(api_config_yaml,
+                                             language,
+                                             repo_root)
+            # TODO(michaelbausor): change default to True once most API and
+            # language combinations are supported
+            if not api_config.get('enable_batch_generation', False):
+                continue
+            api_kwargs = lang_config.copy()
+            api_kwargs.update(api_config)
+            yield api_kwargs
+
+
+def _get_languages(artman_language_yaml, languages):
+    if languages == '*':
+        with open(artman_language_yaml) as config_file:
+            languages = sorted(yaml.load(config_file).keys())
+        if 'common' in languages:
+            languages.remove('common')
+        return languages
+    else:
+        return languages.split(',')
+
+
+def _get_api_configs(api_config_pattern, apis):
```

Comment:
maybe name this _get_artman_configs instead. We have two types of configs - artman & GAPIC; it's hard to determine what "api config" means in relation to those.